contracts/hydro/src/lsm_integration.rs

```diff
@@ -22,16 +21,15 @@ pub const COSMOS_VALIDATOR_ADDR_LENGTH: usize = 52; // e.g. cosmosvaloper15w6ra6
 
 // Returns OK if the denom is a valid IBC denom representing LSM
 // tokenized share transferred directly from the Cosmos Hub
-// of a validator that is also currently among the top
-// max_validators validators, and returns the address of that validator.
+// of a validator that is also among the top max_validators validators
+// for the given round, and returns the address of that validator.
 pub fn validate_denom(
     deps: Deps<NeutronQuery>,
-    env: Env,
+    round_id: u64,
```

Comment:
This is to support denom resolution for the past rounds. We need to pass in the round_id because sometimes this function gets executed in current round, but we are interested for past rounds (e.g. for compounder cap to determine users voting power in previous round).